src/local/room_name.rs

```diff
@@ -0,0 +1,346 @@
+use std::{
+    error,
+    fmt::{self, Write},
+    ops,
+    str::FromStr,
+};
+
+use arrayvec::ArrayString;
+
+/// A structure representing a room name.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub struct LocalRoomName {
+    /// Inner x coordinate representation.
+    ///
+    /// 0 represents E0, positive numbers represent E(x)
+    ///
+    /// -1 represents W0, negative numbers represent W((-x) - 1)
+    pub x_coord: i32,
```

Comment:
Maybe consider leaving those private? This is exposing a lot of the inner structure of something that is meant to be a _name_.

Reply:
Good idea!

I didn't make any changes to the representation of `LocalRoomName` including visibility since I thought that might happen at a later date, but no reason why we can't make it private now.

Reply:
Up to you :)